tests/testthat/test-paginate_listing.R

```diff
@@ -137,18 +137,53 @@ testthat::test_that("checking vertical pagination line calculation.", {
     dplyr::distinct(USUBJID, AGE, BMRKR1, .keep_all = TRUE)
 
   lsting <- as_listing(tmp_data,
-                       key_cols = c("USUBJID", "AGE"),
-                       disp_cols = character(),
-                       main_footer = c("Main Footer A")
+    key_cols = c("USUBJID", "AGE"),
+    disp_cols = character(),
+    main_footer = c("Main Footer A")
   ) %>%
     add_listing_col("BMRKR1", format = "xx.x")
 
   pages_listings <- paginate_listing(lsting, lpp = 8, verbose = TRUE)
 
-  # there is always a gap between the end of the table and the footer. Line calculation is correct given this behaviour
+  # there is always a gap between the end of the table and the footer. Line calculation is correct given this behavior
   page1_result <- matrix_form(pages_listings[[1]])
   page2_result <- matrix_form(pages_listings[[2]])
 
   testthat::expect_equal(sum(nrow(page1_result$strings), length(page1_result$main_footer)), 5)
   testthat::expect_equal(sum(nrow(page2_result$strings), length(page2_result$main_footer)), 5)
 })
+
+testthat::test_that("pagination: lpp and cpp correctly computed for pg_width and pg_height", {
+  lsting <- h_lsting_adae()
+  pag <- paginate_listing(lsting, lpp = 24, cpp = 135)
+  res <- paginate_listing(lsting, pg_width = 15, pg_height = 5)
+  expect_identical(res, pag)
+})
+
+testthat::test_that("pagination: lpp and cpp correctly computed for font_size", {
+  lsting <- h_lsting_adae()
+  pag <- paginate_listing(lsting, lpp = 90, cpp = 105)
+  res <- paginate_listing(lsting, font_size = 8)
+  expect_identical(res, pag)
+})
+
+testthat::test_that("pagination: lpp and cpp correctly computed for lineheight", {
+  lsting <- h_lsting_adae()
+  pag <- paginate_listing(lsting, lpp = 20, cpp = 70)
+  res <- paginate_listing(lsting, lineheight = 3)
+  expect_identical(res, pag)
+})
+
+testthat::test_that("pagination: lpp and cpp correctly computed for landscape", {
+  lsting <- h_lsting_adae()
+  pag <- paginate_listing(lsting, lpp = 45, cpp = 95)
+  res <- paginate_listing(lsting, landscape = TRUE)
+  expect_identical(res, pag)
+})
+
+testthat::test_that("pagination: lpp and cpp correctly computed for margins", {
+  lsting <- h_lsting_adae()
+  pag <- paginate_listing(lsting, lpp = 42, cpp = 65)
```

Comment:
Please add tests that specifies a different page_type other than the default, and a different fontsize than the default as well.

Reply:
It is already there for for non-default `font_size`, the name of the test is: "pagination: lpp and cpp correctly computed for font_size"

Reply:
Ok, so I added one test for non-default `page_type` and `font_size`: "pagination: lpp and cpp correctly computed for page_type and font_size"